apps/rewards/app/store/token.js

```diff
@@ -0,0 +1,198 @@
+import {
+  ETHER_TOKEN_FAKE_ADDRESS,
+  isTokenVerified,
+  tokenDataFallback,
+  getTokenSymbol,
+  getTokenName,
+} from '../utils/token-utils'
+import { addressesEqual } from '../utils/web3-utils'
+import tokenSymbolAbi from '../../../shared/json-abis/token-symbol.json'
+import tokenSymbolBytesAbi from '../../../shared/json-abis/token-symbol-bytes.json'
+import tokenNameAbi from '../../../shared/json-abis/token-name.json'
+import tokenNameBytesAbi from '../../../shared/json-abis/token-name-bytes.json'
+import tokenBalanceAbi from '../../../shared/json-abis/token-balanceof.json'
+import tokenDecimalsAbi from '../../../shared/json-abis/token-decimals.json'
+import { app } from './'
+
+const tokenAbi = [].concat(tokenSymbolAbi, tokenNameAbi, tokenBalanceAbi, tokenDecimalsAbi)
+
+const ETH_CONTRACT = Symbol('ETH_CONTRACT')
+
+export async function initializeTokens(state, settings) {
+  const nextState = {
+    ...state,
+    vaultAddress: settings.vault.address,
+    tokenContracts: new Map(), // Addr -> External contract
+    tokenDecimals: new Map(), // External contract -> decimals
+    tokenNames: new Map(), // External contract -> name
+    tokenSymbols: new Map(), // External contract -> symbol
+  }
+  const withEthBalance = await loadEthBalance(nextState, settings)
+  return withEthBalance
+}
+
+export async function vaultLoadBalance(state, { returnValues }, settings) {
+  const { token } = returnValues
+  const r = await updateBalances(
+    state,
+    token || settings.ethToken.address,
+    settings
+  )
+  return {
+    ...state,
+    balances: r.newBalances,
```

Comment:
I don't like the naming here "balance" over "token" but I don't think to change it everywhere is worth it/this keeps in sync with Aragons naming style but curious on your opinion.

Reply:
I share your views pretty much exactly, Now that it's working I'm leaning towards changing it though